test/chai-sql.coffee

```diff
@@ -101,12 +103,21 @@ exports.operandToAbstractSQLFactory = (binds = [], defaultResource = 'pilot', de
 			fieldParts = operand.split('/')
 			if fieldParts.length > 1
 				alias = parentAlias
-				for fieldPart in fieldParts[...-2]
-					alias = "#{alias}.#{fieldPart.replace(/__/g, '-')}"
-				mapping = clientModel.resourceToSQLMappings[fieldParts[fieldParts.length - 2]][fieldParts[fieldParts.length - 1]]
-				mapping = ["#{alias}.#{mapping[0]}", mapping[1...]...]
+				previousResource = _(parentAlias).split('.').last()
+				for resourceName in fieldParts[...-1]
+					sqlName = resourceNameToSqlName(resourceName)
+					sqlNameParts = sqlName.split('-')
+					mapping = _.get(clientModel.relationships[previousResource], sqlNameParts.join('.'))
+					refTable = mapping[1][0]
+					if sqlNameParts.length > 1 and not _.includes(refTable, '-')
+						# Add the verb to tables that don't include the verb already
+						alias = "#{alias}.#{sqlNameParts[0]}-#{refTable}"
+					else
+						alias = "#{alias}.#{refTable}"
+					previousResource = refTable
+				mapping = [alias, _.last(fieldParts)]
```

Comment:
`_.last(fieldParts)` is an array which is different to what `resourceNameToSqlName(operand)` below seems to be, which means `mapping` can be either of type `[ string, [ string ] ]` or `[ string, string ]` -- is that expected?

Also, you could save `_.last(...)` to a variable above the loop and iterate over that instead of re-accessing the last element in two different ways.